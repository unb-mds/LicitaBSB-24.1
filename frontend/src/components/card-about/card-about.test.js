import React from "react";
import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react";
import CardAbout from ".";

const artigoTeste = {
    primeiroTitulo: "Título teste",
    segundoTitulo: "Subtítulo teste",
    primeiroP: "conteudoTeste1",
    segundoP: "conteudoTeste2",
    urlDaFonte: "linkTeste",
    imagem: "urlTeste",
}

describe("Deve renderizar as seguintes informações do artigo: ", () => {
    test("Título do artigo", () => {
        render(<CardAbout primeiroTitulo={artigoTeste.primeiroTitulo}
            segundoTitulo={artigoTeste.segundoTitulo}
            primeiroP={artigoTeste.primeiroP}
            segundoP={artigoTeste.segundoP}
            urlDaFonte={artigoTeste.urlDaFonte}
            imagem={artigoTeste.imagem} />)
        const info = screen.getByText("Título teste")
        expect(info).toBeInTheDocument(artigoTeste.primeiroTitulo)
    })
    test("Subtítulo do artigo", () => {
        render(<CardAbout primeiroTitulo={artigoTeste.primeiroTitulo}
            segundoTitulo={artigoTeste.segundoTitulo}
            primeiroP={artigoTeste.primeiroP}
            segundoP={artigoTeste.segundoP}
            urlDaFonte={artigoTeste.urlDaFonte}
            imagem={artigoTeste.imagem} />)
        const info = screen.getByText("Subtítulo teste")
        expect(info).toBeInTheDocument(artigoTeste.segundoTitulo)
    })
    test("Primeiro parágrafo do artigo", () => {
        render(<CardAbout primeiroTitulo={artigoTeste.primeiroTitulo}
            segundoTitulo={artigoTeste.segundoTitulo}
            primeiroP={artigoTeste.primeiroP}
            segundoP={artigoTeste.segundoP}
            urlDaFonte={artigoTeste.urlDaFonte}
            imagem={artigoTeste.imagem} />)
        const info = screen.getByText("conteudoTeste1")
        expect(info).toBeInTheDocument(artigoTeste.primeiroP)
    })
    test("Segundo parágrafo do artigo", () => {
        render(<CardAbout primeiroTitulo={artigoTeste.primeiroTitulo}
            segundoTitulo={artigoTeste.segundoTitulo}
            primeiroP={artigoTeste.primeiroP}
            segundoP={artigoTeste.segundoP}
            urlDaFonte={artigoTeste.urlDaFonte}
            imagem={artigoTeste.imagem} />)
        const info = screen.getByText("conteudoTeste2")
        expect(info).toBeInTheDocument(artigoTeste.segundoP)
    })
})