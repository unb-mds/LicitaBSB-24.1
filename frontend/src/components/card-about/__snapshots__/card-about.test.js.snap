// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o snapshot da lista de informações presentes no artigo 1`] = `
Array [
  Array [
    <h2
      class="cardAboutText"
    >
      Título teste
    </h2>,
    <h2
      class="cardAboutText"
    >
      Subtítulo teste
    </h2>,
  ],
  Array [
    <p
      class="cardAboutTextP"
    >
      conteudoTeste1
    </p>,
    <p
      class="cardAboutTextP"
    >
      conteudoTeste2
    </p>,
    <p
      class="cardAboutTextP"
    >
      Para saber mais, acesso este
       
      <a
        href="linkTeste"
        target="_blank"
      >
        link
      </a>
      .
    </p>,
  ],
]
`;
